src/nvim/mouse.c

```diff
@@ -412,6 +412,18 @@ bool do_mouse(oparg_T *oap, int c, int dir, int count, bool fixindent)
 
   // Middle mouse button does a 'put' of the selected text
   if (which_button == MOUSE_MIDDLE) {
+    // handled outside any specific mode
+    if (tab_page_click_defs != NULL) {  // only when initialized
```

Comment:
can/should this live in the other block that checks this same condition, on line 476 (new: 488)